### PR TITLE
elle/BCDA-6025: Smoke Test Refactor: Logs when Err hit in smoke test (bcda_client.go)

### PR DESCRIPTION
**Pull request by 4ell0**
### Fixes [BCDA-6025](https://jira.cms.gov/browse/BCDA-6025)

 Add logging error when err not equal to nil in bcda_client.go

### Proposed Changes

Add logging error when err not equal to nil in bcda_client.go

### Change Details

Add logging error when err not equal to nil in bcda_client.go

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### CODE REVIEW

File: test/smoke_test/bcda_client.go [hunk 1]
```diff
@@ -128,6 +128,7 @@ func startJob(c *client, endpoint, resourceType string) (string, error) {
 
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
+		log.Errorf("Failed to consruct resquest with URL %s", url, err.Error())
```
Comment (krobertson3314) [inline test/smoke_test/bcda_client.go:131]: 'construct request'

File: test/smoke_test/bcda_client.go [hunk 2]
```diff
@@ -136,6 +137,7 @@ func startJob(c *client, endpoint, resourceType string) (string, error) {
 
 	resp, err := c.Do(req)
 	if err != nil {
+		log.Errorf("Exception occured during request call %s", req, err.Error())
```
Comment (krobertson3314) [inline test/smoke_test/bcda_client.go:140]: 'occurred'

File: test/smoke_test/bcda_client.go [hunk 3]
```diff
@@ -156,10 +158,12 @@ func getDataURLs(c *client, jobEndpoint string, timeout time.Duration) ([]string
 	check := func() ([]string, error) {
 		req, err := http.NewRequest("GET", jobEndpoint, nil)
 		if err != nil {
+			log.Errorf("Failure to construt request %s", jobEndpoint, err.Error())
```
Comment (krobertson3314) [inline test/smoke_test/bcda_client.go:161]: 'construct request'

File: test/smoke_test/bcda_client.go [hunk 4]
```diff
@@ -156,10 +158,12 @@ func getDataURLs(c *client, jobEndpoint string, timeout time.Duration) ([]string
...
 			return nil, err
 		}
 		resp, err := c.Do(req)
 		if err != nil {
+			log.Errorf("Exception occured during call %s", req, err.Error())
```
Comment (krobertson3314) [inline test/smoke_test/bcda_client.go:166]: 'occurred'

File: test/smoke_test/bcda_client.go [hunk 5]
```diff
@@ -218,17 +225,20 @@ func getDataURLs(c *client, jobEndpoint string, timeout time.Duration) ([]string
 func getData(c *client, dataURL string) ([]byte, error) {
 	req, err := http.NewRequest("GET", dataURL, nil)
 	if err != nil {
+		log.Errorf("Failure to construct request %s", dataURL, err.Error())
 		return nil, err
 	}
 	req.Header.Add("Accept-Encoding", "gzip")
 	resp, err := c.Do(req)
 	if err != nil {
+		log.Errorf("Exception occured when making the call %s", req, err.Error())
```
Comment (krobertson3314) [inline test/smoke_test/bcda_client.go:234]: 'occurred'

File: test/smoke_test/bcda_client.go [hunk 6]
```diff
@@ -316,6 +329,7 @@ func (c *client) updateAccessToken() error {
 	// retry loop we dont want to use the retry logic on updating the access token
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
+		log.Errorf("Exception occured when making request call %s", req, err.Error())
```
Comment (krobertson3314) [inline test/smoke_test/bcda_client.go:332]: 'occurred'

**Review by famegashie — APPROVED** [review]
Reviewed.

**Pull request merged**